src/public/pages/main.js

```diff
@@ -26,8 +26,12 @@ const Contacts = ({contacts}) =>
     Number of contacts: {U.length(contacts)}
   </div>
 
-export default U.withContext((_, {state}) =>
+export default U.withContext((props, {state}) =>
   <div>
     <h1>Main page</h1>
     <Contacts contacts={U.view(["contacts", L.define([])], state)}/>
-  </div>)
+    { typeof L.get( [ 'params', 'id' ], props ) !== 'undefined'
+      && <div>With ID: { L.get( [ 'params', 'id' ], props ) }</div>
+    }
```

Comment:
Karet Util has functions `ift` (aka "if-then") and `ifte` (aka "if-then-else") for conditionals like this.  You could write:

```js
{ U.ift(L.get( [ 'params', 'id' ], props ),
        <div>With ID: { L.get( [ 'params', 'id' ], props ) }
```

Note that with `undefined` there is no need to perform a `typeof` test.

I would also suggest dropping the separate `params` object and just matching the `id` property directly which would simplify this whole thing to:

```js
export default U.withContext(({id}, {state}) =>
  <div>
    <h1>Main page</h1>
    <Contacts contacts={U.view(["contacts", L.define([])], state)}/>
    { U.ift(id, <div>With ID: { id }</div> }
  </div>
)
```